### ISSUE TITLE
Auto Maps Getting 'Stuck' in Map

### ISSUE THREAD

**Issue by kevrlet** [zininzinin/AutoTrimps#86]
I've noticed it before, but now that I'm pushing deeper it's happening in every zone. With Auto Maps on, the script is looping on a map over 100 times despite having no reason to; if I turn off Auto Maps and manually go back to World, I clear not only the zone I'm on but often the next 1 or 2 zones.

Probably an issue with calculating survival chances against the strongest possible imp in a zone?

**Comment by spindrjr** [comment]
This may be expected behavior.
Since it always tries to keep you in 1-shot range on maps, if you are pushing very deep past efficient he/hr areas, there will come a point where you are far behind on coordination and it's spending a lot of time farming to try to get upgrades to keep damage up.

You can turn off automaps and manually push ahead a few zones, but you will quickly run out of steam (and probably be out of 1-shot map range when you do). Since I just recently added siphonology logic, there may be a a few zones wiggle room that's not accounted for by the difficulty check, but it sounds like you're in this general area.

TLDR: The goal of the automaps algorithms is to efficiently farm you helium, not push you deep, so you may have to do some stuff yourself when push.

If you think it's not behaving like this, I'd be happy to take a look at an export.

**Comment by kevrlet** [comment]
If that's how it's meant to work, you're probably right and that's probably
what it is. I just assumed it was supposed to push you over the 'you can't
survive this zone' hump before going back to World, and to cut out early if
you hit the point of one-shotting everything on the World map, because
that's what I think when I see 'Auto Map'.

Sorry for the trouble :)

**Comment by spindrjr** [comment]
No trouble. Yea it definitely does not push the limits like that. This is half me leaving some game to the players, and half avoiding the (I think) tricky programming. The fact that everyone has different standards for acceptable clear speed while pushing just compounds it.

I appreciate your attention to detail though in looking for issues. Please continue!
